src/nvim/regexp_nfa.c

```diff
@@ -6064,10 +6070,17 @@ static int nfa_regmatch(nfa_regprog_T *prog, nfa_state_T *start, regsubs_T *subm
       break;
 
     // Allow interrupting with CTRL-C.
-    fast_breakcheck();
+    line_breakcheck();
```

Comment:
@jbradaric, I just run `make unittest` or `TEST_FILTER="should prefer to match the start of a file path" make unittest`. 

Could you replace `line_breakcheck()` with the direct call to `os_breakcheck()` and see if this triggeres the problem for you?